File: predictions/app/src/main/java/com/google/firebase/example/predictions/PreventChurnActivity.java
package com.google.firebase.example.predictions;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.gms.tasks.Continuation;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.remoteconfig.FirebaseRemoteConfig;

import java.util.HashMap;
import java.util.Map;

public class PreventChurnActivity extends AppCompatActivity {

    private void preventChurn() {
        // [START pred_prevent_churn]
        final FirebaseRemoteConfig config = FirebaseRemoteConfig.getInstance();

        Map<String, Object> remoteConfigDefaults = new HashMap<>();
        remoteConfigDefaults.put("grant_retention_gift", false);
        config.setDefaultsAsync(remoteConfigDefaults)
                .continueWithTask(new Continuation<Void, Task<Boolean>>() {
                    @Override
                    public Task<Boolean> then(@NonNull Task<Void> task) throws Exception {
                        if (!task.isSuccessful()) {
                            throw task.getException();
                        }
                        return config.fetchAndActivate();
                    }
                })
                .addOnCompleteListener(this, new OnCompleteListener<Boolean>() {
                    @Override
                    public void onComplete(@NonNull Task<Boolean> task) {
                        if (task.isSuccessful()) {
                            // Act on the retrieved parameters. For example, grant the
                            // retention gift to players who haven't yet received one.
                            boolean shouldGrantGift = config.getBoolean("grant_retention_gift");
                            if (shouldGrantGift && !playerAlreadyReceivedGift()) {
                                grantGiftToPlayer();
                            }
                        }
                    }
                });
        // [END pred_prevent_churn]
    }

    private boolean playerAlreadyReceivedGift() {
        // Dummy method
        return false;
    }

    private void grantGiftToPlayer() {
        // Dummy method
    }

}

File: predictions/app/src/main/java/com/google/firebase/example/predictions/kotlin/PreventChurnActivity.kt
package com.google.firebase.example.predictions.kotlin

import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.ktx.Firebase
import com.google.firebase.remoteconfig.ktx.get
import com.google.firebase.remoteconfig.ktx.remoteConfig

class PreventChurnActivity : AppCompatActivity() {

    private fun preventChurn() {
        // [START pred_prevent_churn]
        val config = Firebase.remoteConfig

        val remoteConfigDefaults = hashMapOf<String, Any>(
                "grant_retention_gift" to false
        )
        config.setDefaultsAsync(remoteConfigDefaults)
                .continueWithTask { task ->
                    if (!task.isSuccessful) {
                        task.exception?.let {
                            throw it
                        }
                    }
                    config.fetchAndActivate()
                }
                .addOnCompleteListener(this) { task ->
                    if (task.isSuccessful) {
                        // Act on the retrieved parameters. For example, grant the
                        // retention gift to players who haven't yet received one.
                        val shouldGrantGift = config["grant_retention_gift"].asBoolean()
                        if (shouldGrantGift && !playerAlreadyReceivedGift()) {
                            grantGiftToPlayer()
                        }
                    } else {
                        // Handle errors
                    }
                }
        // [END pred_prevent_churn]
    }

    private fun playerAlreadyReceivedGift(): Boolean {
        // Dummy method
        return false
    }

    private fun grantGiftToPlayer() {
        // Dummy method
    }
}
